internal/controller/clusterdeployment_controller.go

```diff
@@ -754,6 +761,7 @@ func (r *ClusterDeploymentReconciler) reconcileDelete(ctx context.Context, cd *k
 		return ctrl.Result{RequeueAfter: r.defaultRequeueTime}, nil
 	}
 	if !apierrors.IsNotFound(err) {
+		r.setCondition(cd, kcm.DeletingCondition, err)
```

Comment:
in that case, i'd say we have to set it to true/ready if no errors just before finalizer deletion